File: demo/RFID/rfid/app/models/status/pick_status.rb
class PickStatus<BaseStatus
  INIT = 100
  PICKING = 200
  PICKED = 300
  ABORTED = 400

  def self.display state
    case state
      when INIT
        '初始化'
      when PICKING
        '进行中'
      when PICKED
        '完成'
      else
        '放弃'
    end
  end
end

File: programme/wms/windows/ScmClient/ScmWcfService/Model/Enum/PickStatus.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ScmWcfService.Model.Enum
{
    public enum PickStatus
    {
        INIT=100,
        PICKING=200,
        PICKED=300,
        ABORTED=400
    }

    public class PickStatusDisplay {
        public static string GetStatusOfPick(PickStatus status) {
            string result = string.Empty;
            switch (status) { 
                case PickStatus.INIT:
                    result = "NEW";
                    break;
                case PickStatus.PICKING:
                    result = "PICKING";
                    break;
                case PickStatus.PICKED:
                    result = "PICKED";
                    break;
                case PickStatus.ABORTED:
                    result = "ABORTED";
                    break;
                default:
                    result = "N/A";
                    break;
            }
            return result;
        }
    }
}